Sources/StytchUI/StytchB2BUIClient/Shared/BaseViewController/BaseViewController+MFA.swift

```diff
@@ -33,7 +29,7 @@ extension BaseViewController {
                 }
             }
 
-            if let viewController {
+            if let viewController, B2BAuthenticationManager.isAuthenticated == false {
                 navigationController?.pushViewController(viewController, animated: true)
```

Comment:
This will prevent the flicker to the next view controller when the user is fully authenticated.